### ISSUE TITLE
Inline comments to push/pop state instead of replacing it

### ISSUE THREAD

**Issue by jeremykentbgross** [eslint/eslint#18268]
### ESLint version

8.48.0

### What problem do you want to solve?

The ability to change rule state inline with `/* eslint-disable-line <some rule> */` and `/* eslint-disable-next-line <some rule> */` are really handy for special edge cases, but it has limitations.

After running my obfuscation/minification, (which can preserve the eslint directives), I run eslint for a second time to make sure nothing broke in that process.  At this point, many lines are converged together, and disabling the rule for the line, or the next line, affects more code, or possibly different code, than it was intended for.

(Note: I also have a some multiline macro system that the current functionality is difficult with, but I won't bother you with those details since my proposed solution solves both problems.)

I could use `/* eslint-disable <some rule> */` `/* eslint-enable <some rule> */`, but as I understand it, that may cause the enabling of rules that were not on in the first place from my top level configuration (which varies per build type), or change the state of them when re-enabling them if they had more detailed settings than just being on or off.

### What do you think is the correct solution?

I therefore propose a new inline mechanic that would push/pop the state of a rule to allow returning to the previous state exactly (regardless of what it's previous state was), that could look something like this:

```
/* eslint-disable-push <some rule> */
//...
//    <some code block>
//...
/* eslint-pop <some rule> */


/* eslint-disable-push <some rule> */    <...some code...>    /* eslint-pop <some rule> */
...
```

```
****
TL;DR: The ability to have comments push/pop state instead of just replacing it would be a big improvement.
****
```

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

I noticed that right above this box is a checkbox for if I would submit a change for this.  At present I don't know the first thing about the eslint code base, or how long this would take me given I have little knowledge of it's internals or complexity, but at present I would like to state the checkbox to be in a super position of both on and off.

That is, if someone familiar with the code base can recommend the sort of approach that would likely be acceptable for a submitted CL, I could consider trying my hand at it, depending on the complexity, and how much trouble the lack of this feature gives me in the near future.  This would pull time away from other deadlines I have, but I am saying that the lack of the ticked box does not mean my stance is a hard no, but a maybe.

**Comment by mdjermanovic** [comment]
> I could use `/* eslint-disable <some rule> */` `/* eslint-enable <some rule> */`, but as I understand it, that may cause the enabling of rules that were not on in the first place from my top level configuration (which varies per build type), or change the state of them when re-enabling them if they had more detailed settings than just being on or off.

`/* eslint-enable <some rule> */` just re-enables reporting from the rule that was disabled with `/* eslint-disable <some rule> */`. If the rule isn't in the configuration, `/* eslint-enable <some rule> */` won't have any effects.

[Playground Demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50LWRpc2FibGUgbm8tdW5kZWYgKi9cbi8qIGVzbGludC1lbmFibGUgbm8tdW5kZWYgKi9cblxuZm9vKCk7IC8vIG5vIGVycm9ycyBoZXJlIGJlY2F1c2UgdGhlIHJ1bGUgaXNuJ3QgaW4gdGhlIGNvbmZpZ3VyYXRpb25cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)

Do you have an example where this doesn't work for your use case?
